lang/well-formed.md

```diff
@@ -19,6 +19,8 @@ fn ensure(b: bool) -> Option<()> {
 impl IntType {
     fn check(self) -> Option<()> {
         ensure(self.size.bytes().is_power_of_two())?;
+        // We only support int types that are no larger than 128 bits.
```

Comment:
This is what rustc currently does, but is there anything that would break in Minirust if things would become bigger?

For now I am inclined to permit arbitrary integer sizes here; that doesn't mean the rustc frontend has to use them all.

Reply:
Considering only the integer types, there's no problem with supporting arbitrary-sized integers. But if we support arbitrary-sized integers, should we support arbitrary-sized floats? (Though MiniRust doesn't support floats yet)

Reply:
Rust doesn't have `f16` or `f128` so I think the answer to your question is 'no'.

But anyway MiniRust should treat floats very abstractly I think, via some softloat type provided as an external library. floats are pretty low -priority right now though...

Reply:
Indeed, floats are much more complicated than integers. It lets me think of the `long double` in C++, which might be `f80` in Rust (the size is even not the power of two).

If we treat floats completely differently from integers (i.e., not by encoding/decoding from raw bytes and reinterpreting the bits), I agree not to limit the size of integer types. It would be a superset of the integer types that rustc supports.